src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/FlowWithSubFlowsEntityPath.groovy

```diff
@@ -0,0 +1,106 @@
+                        reverse: new Path(flowWithSubFlowsPaths.sharedPath.protectedPath.reverse, topologyDefinition)
+                ))
+
+        this.subFlowPaths = flowWithSubFlowsPaths.subFlowPaths.collect { subFlow ->
+            new FlowPathModel(
+                    flowId: subFlow.flowId,
+                    path: new PathModel(
+                            forward: new Path(subFlow.forward, topologyDefinition),
+                            reverse: new Path(subFlow.reverse, topologyDefinition)
+                    ),
+                    protectedPath: !subFlow?.protectedPath ? null : new PathModel(
+                            forward: new Path(subFlow.protectedPath.forward, topologyDefinition),
+                            reverse: new Path(subFlow.protectedPath.reverse, topologyDefinition)
+                    )
+            )
+        }
+
+        this.topologyDefinition = topologyDefinition
+    }
+
+    FlowWithSubFlowsEntityPath(YFlowPaths flowWithSubFlowsPaths, TopologyDefinition topologyDefinition) {
+        this.sharedPath = new FlowPathModel(
+                path: new PathModel(
+                        forward: new Path(flowWithSubFlowsPaths.sharedPath.forward, topologyDefinition),
+                        reverse: new Path(flowWithSubFlowsPaths.sharedPath.reverse, topologyDefinition),
+                        diverseGroup: flowWithSubFlowsPaths.sharedPath.diverseGroup
+                ),
+                protectedPath: !sharedPath?.protectedPath ? null : new PathModel(
+                        forward: new Path(flowWithSubFlowsPaths.sharedPath.protectedPath.forward, topologyDefinition),
+                        reverse: new Path(flowWithSubFlowsPaths.sharedPath.protectedPath.reverse, topologyDefinition),
+                        diverseGroup: flowWithSubFlowsPaths.sharedPath.protectedPath.diverseGroup
+                ))
+
+        this.subFlowPaths = flowWithSubFlowsPaths.subFlowPaths.collect { subFlow ->
+            new FlowPathModel(
+                    flowId: subFlow.flowId,
+                    path: new PathModel(
+                            forward: new Path(subFlow.forward, topologyDefinition),
+                            reverse: new Path(subFlow.reverse, topologyDefinition),
+                            diverseGroup: subFlow.diverseGroup
+
+                    ),
+                    protectedPath: !subFlow?.protectedPath ? null : new PathModel(
+                            forward: new Path(subFlow.protectedPath.forward, topologyDefinition),
+                            reverse: new Path(subFlow.protectedPath.reverse, topologyDefinition),
+                            diverseGroup: subFlow.diverseGroup
+                    )
+            )
+        }
+
+        this.topologyDefinition = topologyDefinition
+    }
+
+    List<Isl> getInvolvedIsls(boolean isForward = true) {
+        subFlowPaths.collect { it.getInvolvedIsls(isForward) }.flatten().unique() as List<Isl>
```

Comment:
I think in this particular part we should return Set. 
In case of simple flow, returning list makes sense because we have a clear straight order of ISLs. But in case of y- and ha-flows we don't have an order behind shared part.

Reply:
`List<Isl>` was used as methods to manipulate with ISLs(set up bandwidth/cost/etc) is for this type. We can overload existing methods as a part of another PR.
As for the order, if I am not mistaken, we have the same structure as for the regular flow due to collecting ISL per subflow but then we merge each subflow's ISLs.